pkg/shortnames/shortnames.go

```diff
@@ -386,6 +384,55 @@ func Resolve(ctx *types.SystemContext, name string) (*Resolved, error) {
 	return resolved, nil
 }
 
+// canPromptWithTTY indicates if stdout or stdin is not a TTY.
+func canPromptWithTTY() bool {
+	return term.IsTerminal(int(os.Stdout.Fd())) && term.IsTerminal(int(os.Stdin.Fd()))
+}
+
+// Prompt opens a prompt in the terminal to let the user choose *one* of the
+// specified items.
+func Prompt(mode types.ShortNameMode, original string, items []string) (string, error) {
+	switch len(items) {
+	case 0:
+		return "", fmt.Errorf("cannot prompt without items")
+	case 1:
+		// NOTE: return the original will make sure that information on
+		// how the (possibly short) name is resolved afterwards. For
+		// instance, `podman-remote pull short` will then print if
+		// `short` resolved to an alias.
+		//
+		// Returning `items[0]` would return a FQN and hence hide the
+		// fact that it's an alias when actually pulling down the
+		// image.
+		return original, nil
+	default:
+		if !canPromptWithTTY() {
+			if mode == types.ShortNameModeEnforcing {
+				return "", errNoTTY
+			}
+			return original, nil
```

Comment:
OK, that’s sort of an argument to support returning `original`, but what’s the caller going to do when the remote server refuses to prompt locally, and the local client can’t prompt?

(I wouldn’t mind `Prompt` not dealing with `mode` at all, but that just moves the conversation to another PR.)

How can the code path returning `original` succeed?

Reply:
The passed in `mode` is the one from the server.  Since we already now it's not enforcing, the server would then fall back to walking the search registries.